### PR TITLE
chore: streamlining local docker integration

### PR DESCRIPTION
**Pull request by arboleya**
Aiming at uncluttering the repo _root_ dir and improving _nomenclatures_, this PR:

- Deletes `<repoRoot>/Makefile` while porting its commands to `package.json`
- Relocates `<repoRoot>/{services,docker-compose.yml}` to `<repoRoot>/.docker` for increased clarity

> **Note** Extracted from #827

### CODE REVIEW

**Review by Dhaiwat10 — APPROVED** [review]
Super clean. ✨

**Review by danielbate — APPROVED** [review]

**Review by Torres-ssf — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -21,15 +21,21 @@
...
     "changeset:version-with-docs": "ts-node ./scripts/changeset-version-with-docs",
     "forc:update": "ts-node ./scripts/forc-update",
     "forc:check": "./scripts/forc-check.sh",
     "forc:format": "./scripts/forc-format.sh",
+    "docker:run": "cd ./.docker/fuel-core && docker compose up --build",
+    "docker:run:bg": "pnpm docker:run -d",
+    "docker:clean": "run-s docker:clean:*",
+    "docker:clean:containers": "docker ps --filter name=fuel-core* -aq | xargs docker stop | xargs docker rm -f",
+    "docker:clean:images": "docker image ls --filter \"reference=fuel-core*\" -q | xargs docker rmi -f",
+    "docker:clean:volumes": "docker volume ls --filter name=fuel-core* -q | xargs docker volume rm -f",
```
[the remark this answers is not in the document]
Comment (danielbate) [reply package.json:38]: I agree with keeping the `package.json` tidy, but still think there is value in documenting these _somewhere_. Perhaps a new docker section within the [contribution documentation](https://github.com/FuelLabs/fuels-ts/blob/master/CONTRIBUTING.md).

File: .docker/fuel-core/Dockerfile [hunk 2]
```diff
@@ -5,6 +5,10 @@ ARG PORT=4000
 ARG MIN_GAS_PRICE=0
 ARG DB_PATH=./mnt/db/
 
+# This is the private key of the consensus.PoA.signing_key in the chainConfig.json
+# this key is responsible for validating the transactions
+ARG CONSENSUS_KEY=0xa449b1ffee0e2205fa924c6740cc48b3b473aa28587df6dab12abc245d1f5298
+
 ENV IP="${IP}"
 ENV PORT="${PORT}"
 ENV DB_PATH="${DB_PATH}"
```
[the remark this answers is not in the document]
Comment (arboleya) [reply .docker/fuel-core/Dockerfile]: It doesn't work without the `ENV`, unfortunately. `ARG`s can only be used a build time, but we need these values in runtime during `CMD` — hence why we need `ENV`. However, `ARG`s can be removed: 7b0a59e9b03c8196b7848c0d32471272142d6f59.

File: .docker/fuel-core/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+version: "3"
+
+services:
+  fuel-core:
```
[the remark this answers is not in the document]
Comment (arboleya) [reply .docker/fuel-core/docker-compose.yml:4]: 807c6e82a42d83be3a477f6f771de03d21ad5373

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (arboleya) [reply package.json:38]: I can think of a few things:
 1. IIRC, whenever someone updates the `fuel-core`, we need to delete the old docker images _manually_. That's where the image removal comes in.
 2. Also, I remember having found these scripts very useful when upgrading `fuel-core` myself. During debugging, I'd need to switch local branches and trees with different Dockerfiles. Using a single command to remove and rebuild everything from scratch was very practical.
 3. Finally, these scripts are all used by the `test:ci` command. We need a clear environment for running all tests locally, and these commands will run to ensure that.

I agree with the undesirable extra weight to the `package.json` file, but for now, I don't see any harm in it compared to its utility. We can think of better segmenting these things later on.

**Review by danielbate — APPROVED** [review]

**Review by Dhaiwat10 — APPROVED** [review]

**Pull request merged**